assets/js/omniSelectEngine.js

```diff
@@ -0,0 +1,394 @@
+$(document).ready(function ($) {
+  const userId = omniSearch.settings.userId;
+  const elementIcons = {
+    task: "fa fa-fw fa-tasks",
+    project: "fa fa-fw fa-shapes",
+  };
+  // Append overlay
+  $("body").append(`
+      <div class="omni-search hidden">
+          <select class="js-example-basic-multiple" name="actions[]"></select>
+      </div>`);
+  const omniSelectElement = $("body .omni-search > select");
+
+  $("div.omni-search").on("click", function (e) {
+    // Close overlay when clicking outside.
+    if ($(e.target).hasClass("omni-search")) {
+      destroyOmniSearch();
+    }
+  });
+  $("div.content-container").on("click", function (e) {
+    // Close overlay when clicking outside (when logging time).
+    if ($(e.target).hasClass("content-container")) {
+      destroyOmniSearch();
+    }
+  });
+  // Event for init and destroy
+  $("body").on("keydown", function (e) {
+    const keyCode = e.keyCode;
+    switch (keyCode) {
+      case 27: // escape key.
+        destroyOmniSearch();
+        break;
+      case 190: // dot key.
+        if (!$("input, textarea").is(":focus")) {
+          initOmniSearch();
+        }
+        break;
+    }
+  });
+
+  // Init select2, get data, set events.
+  function initOmniSearch() {
+    $("body").addClass("prevent-scroll");
+    omniSelectElement.addClass("loading");
+    if ($(".omni-search").hasClass("hidden") === false) {
+      return false;
+    }
+
+    $.fn.select2.amd.require(["select2/selection/search"], function (Search) {
+      var oldRemoveChoice = Search.prototype.searchRemoveChoice;
+
+      Search.prototype.searchRemoveChoice = function () {
+        oldRemoveChoice.apply(this, arguments);
+        this.$search.val("");
+      };
+
+      // Init select2
+      omniSelectElement
+        .select2({
+          multiple: true,
+        })
+        .on("select2:select", function (e) {
+          const data = e.params.data;
+          $(".select2-search__field").val(data.text);
+          switch (data.type) {
+            case "project":
+              $(".selected-value").text(data.text);
+              reinitOmniSearchForType("project", data);
+              break;
```

Comment:
I think that having an empty line after `break;` makes the code less compact and more readable.

Reply:
Did this!